### PR TITLE
shields: mcp2515: add support for Keyestudio CAN-BUS shield

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for the Keyestudio CAN-BUS shield (KS0411). This shield is very similar to the DFRobot CAN-BUS shield v2.0.

Rename the MCP2515 devicetree nodes in order not to conflict with on-chip CAN controllers.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

**Review by martinjaeger — DISMISSED** [review]
Looks good.

File: boards/shields/mcp2515/keyestudio_can_bus_ks0411.overlay [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		spi-max-frequency = <1000000>;
+		int-gpios = <&arduino_header 14 GPIO_ACTIVE_LOW>; /* D8 */
+		status = "okay";
+		label = "MCP2515";
+		reg = <0x0>;
+		osc-freq = <16000000>;
+		bus-speed = <125000>;
+		sjw = <1>;
+		prop-seg = <2>;
+		phase-seg1 = <7>;
```
Comment (alexanderwachter) [inline boards/shields/mcp2515/keyestudio_can_bus_ks0411.overlay:22]: Shouldn't we use the sample point instead?

Comment (henrikbrixandersen) [reply boards/shields/mcp2515/keyestudio_can_bus_ks0411.overlay]: Yes. I intend to do a full sweep of the tree and replace it all over. I have changed it for the MCP2515 shields now.

**Review by alexanderwachter — APPROVED** [review]

**Review by erwango — APPROVED** [review]
Would have been nice for the review to split the changes a bit.
Anyway LGTM

**Review by jfischer-no — APPROVED** [review]
:+1:

**Review by mbolivar-nordic — APPROVED** [review]

**Review by karstenkoenig — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

**Pull request merged**